### PR TITLE
Prevent binary data in log

### PR DESCRIPTION
**Pull request by keliramu**

### CODE REVIEW

File: events/logger/logger.go [hunk 1]
```diff
@@ -118,11 +119,16 @@ func dataRequestAPIToString(
 		b.WriteString(fmt.Sprintf("Error: %s\n", data.Error))
 	}
 	if data.Response != nil {
+		tmpBody := ""
+		// do not print binary data
+		if !slices.Contains(data.Response.Header.Values("Content-Type"), "application/octet-stream") {
```
Comment (mariusSincovici) [inline events/logger/logger.go:124]: [small] if we'll still have issues or the responses are not correctly marked as binary, maybe `utf8.validate` can be used.

**Review by mariusSincovici — APPROVED** [review]

File: events/logger/logger.go [hunk 2]
```diff
@@ -118,11 +119,16 @@ func dataRequestAPIToString(
 		b.WriteString(fmt.Sprintf("Error: %s\n", data.Error))
 	}
 	if data.Response != nil {
+		tmpBody := ""
```
Comment (devzbysiu) [inline events/logger/logger.go:122]: [small] maybe let's set it to `"(binary data)" just to indicate that there is something but it's binary?

**Review by devzbysiu — APPROVED** [review]

File: events/logger/logger.go (hunk 1 above)
Comment (keliramu) [reply events/logger/logger.go:124]: How expensive is `utf8.validation`? Extra validation/processing wastes resources - is it worth doing it here? How many cases you know with not correctly marked binary in http response?

File: events/logger/logger.go (hunk 2 above)
Comment (keliramu) [reply events/logger/logger.go:122]: good idea, yes, I'll add this

**Pull request merged**